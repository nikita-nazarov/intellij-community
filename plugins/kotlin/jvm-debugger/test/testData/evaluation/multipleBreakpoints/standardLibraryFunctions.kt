package standardLibraryFunctions

fun main() {
    val list = listOf(1, 2, 3)
    list
        //Breakpoint! (lambdaOrdinal = 1)
        .map { it / 2 }
        //Breakpoint! (lambdaOrdinal = 1)
        .filter { it > 1 }
}

// EXPRESSION: it
// RESULT: 1: I
// EXPRESSION: it
// RESULT: 2: I
// EXPRESSION: it
// RESULT: 3: I

// EXPRESSION: it
// RESULT: 0: I
// EXPRESSION: it
// RESULT: 1: I
// EXPRESSION: it
// RESULT: 1: I

// PRINT_FRAME
// SHOW_KOTLIN_VARIABLES
