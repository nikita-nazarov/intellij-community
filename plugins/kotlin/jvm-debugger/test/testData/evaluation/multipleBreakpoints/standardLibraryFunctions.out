LineBreakpoint created at standardLibraryFunctions.kt:7 lambdaOrdinal = 1
LineBreakpoint created at standardLibraryFunctions.kt:9 lambdaOrdinal = 1
Run Java
Connected to the target VM
standardLibraryFunctions.kt:7
Compile bytecode for it
InlineStackFrame lambda 'map' in 'main' (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 1
InlineStackFrame mapTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 0
    JavaValue[local] item: java.lang.Object = 1
        JavaValue[field] value: int = 1 (Integer.!EXT!)
InlineStackFrame map (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
standardLibraryFunctions.kt:7
InlineStackFrame lambda 'map' in 'main' (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 2
InlineStackFrame mapTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 1
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
    JavaValue[local] item: java.lang.Object = 2
        JavaValue[field] value: int = 2 (Integer.!EXT!)
InlineStackFrame map (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
standardLibraryFunctions.kt:7
InlineStackFrame lambda 'map' in 'main' (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 3
InlineStackFrame mapTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 2
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
    JavaValue[local] item: java.lang.Object = 3
        JavaValue[field] value: int = 3 (Integer.!EXT!)
InlineStackFrame map (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:7)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
standardLibraryFunctions.kt:9
Compile bytecode for it
InlineStackFrame lambda 'filter' in 'main' (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 0
InlineStackFrame filterTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 0
    JavaValue[local] element: java.lang.Object = 0
        JavaValue[field] value: int = 0 (Integer.!EXT!)
InlineStackFrame filter (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
standardLibraryFunctions.kt:9
InlineStackFrame lambda 'filter' in 'main' (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 1
InlineStackFrame filterTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 0
    JavaValue[local] element: java.lang.Object = 1
        JavaValue[field] value: int = 1 (Integer.!EXT!)
InlineStackFrame filter (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
standardLibraryFunctions.kt:9
InlineStackFrame lambda 'filter' in 'main' (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
    JavaValue[local] it: int = 1
InlineStackFrame filterTo (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
    JavaValue[local] destination: java.util.Collection =  size = 0
    JavaValue[local] element: java.lang.Object = 1
        JavaValue[field] value: int = 1 (Integer.!EXT!)
InlineStackFrame filter (_Collections.!EXT!)
    JavaValue[local] this: java.lang.Iterable =  size = 3
        JavaValue[element] 0: Integer@uniqueID = 0
            JavaValue[field] value: int = 0 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
KotlinStackFrame (standardLibraryFunctions.kt:9)
    JavaValue[local] list: java.util.List =  size = 3 (standardLibraryFunctions.kt:4)
        JavaValue[element] 0: Integer@uniqueID = 1
            JavaValue[field] value: int = 1 (Integer.!EXT!)
        JavaValue[element] 1: Integer@uniqueID = 2
            JavaValue[field] value: int = 2 (Integer.!EXT!)
        JavaValue[element] 2: Integer@uniqueID = 3
            JavaValue[field] value: int = 3 (Integer.!EXT!)
Disconnected from the target VM

Process finished with exit code 0
